src/key.rs

```diff
@@ -1216,6 +1215,26 @@ impl FromStr for Key {
     }
 }
 
+#[cfg(feature = "serde")]
+impl serde::Serialize for Key {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&format!("{}", self))
```

Comment:
Consider using `serializer.collect_str(self)` so the Display impl writes straight into the output without making a String.